src/main/java/org/folio/consortia/service/impl/SyncPrimaryAffiliationServiceImpl.java

```diff
@@ -70,30 +81,50 @@ private SyncPrimaryAffiliationBody buildSyncPrimaryAffiliationBody(String tenant
   @Override
   public void createPrimaryUserAffiliations(UUID consortiumId, String centralTenantId,
     SyncPrimaryAffiliationBody syncPrimaryAffiliationBody) {
-    log.info("Start creating user primary affiliation for tenant {}", syncPrimaryAffiliationBody.getTenantId());
-    var tenantId = syncPrimaryAffiliationBody.getTenantId();
-    var userList = syncPrimaryAffiliationBody.getUsers();
-    TenantEntity tenantEntity = tenantService.getByTenantId(tenantId);
+    try {
+      log.info("Start creating user primary affiliation for tenant {}", syncPrimaryAffiliationBody.getTenantId());
+      var tenantId = syncPrimaryAffiliationBody.getTenantId();
+      var userList = syncPrimaryAffiliationBody.getUsers();
+      TenantEntity tenantEntity = tenantService.getByTenantId(tenantId);
+      createPrimaryUserAffiliations(consortiumId, centralTenantId, tenantId, userList, tenantEntity);
+    } catch (Exception e) {
+      log.error("createPrimaryUserAffiliations:: error creating user primary affiliations", e);
+      tenantService.updateTenantSetupStatus(syncPrimaryAffiliationBody.getTenantId(), centralTenantId, SetupStatusEnum.FAILED);
+      throw e;
+    }
+  }
+
+  private void createPrimaryUserAffiliations(UUID consortiumId, String centralTenantId, String tenantId,
+    List<SyncUser> userList, TenantEntity tenantEntity) {
+    var affiliatedUsersCount = new AtomicInteger(0);
+    var hasFailedAffiliations = new AtomicBoolean(false);
```

Comment:
Basically these variables are local scoped, so there is no any issues with concurrent access here, we can use also common  counters

Reply:
Update: atomic variables using here to bypass making varaibles final to use them in lambda.